File: internal/domain/markdown_test.go
package domain

import "testing"

func TestRenderMarkdown(t *testing.T) {
	c := Cluster{
		ID:           "cluster_1",
		CanonicalURL: "https://example.com/a",
		Title:        "Topic A",
		RankScore:    1,
		Items:        []ScoredItem{{SourceItem: SourceItem{Source: SourceHackerNews, ID: "1", Title: "Topic A", URL: "https://example.com/a"}, RankScore: 1}},
	}
	md := RenderMarkdown("2026-03-02", "バックエンドエンジニア", []Cluster{c}, []EnrichedCluster{{ClusterID: "cluster_1", IsImportant: true, SummaryJA: "要約", ReasonToRead: "理由"}})
	if md == "" || !contains(md, "Topic A") || !contains(md, "URL: https://example.com/a") || !contains(md, "要約") {
		t.Fatalf("unexpected markdown: %s", md)
	}
}

func contains(s, sub string) bool { return len(s) >= len(sub) && (indexOf(s, sub) >= 0) }

func indexOf(s, sub string) int {
	for i := 0; i+len(sub) <= len(s); i++ {
		if s[i:i+len(sub)] == sub {
			return i
		}
	}
	return -1
}

File: test/domain/markdown.test.ts
import { describe, expect, it } from "vitest";
import { renderMarkdown } from "../../src/domain/markdown.js";
import { Cluster } from "../../src/domain/model.js";

function makeCluster(id: string, title: string): Cluster {
  return {
    id,
    canonicalUrl: `https://example.com/${id}`,
    title,
    trendScore: 2,
    deltaScore: 5,
    deltaComments: 3,
    items: [
      {
        source: "hackernews",
        id: `${id}-item`,
        title,
        url: `https://example.com/${id}`,
        score: 10,
        comments: 3,
        publishedAt: "2026-03-02T00:00:00.000Z",
        ageHours: 1,
        deltaScore: 5,
        deltaComments: 3,
        trendScore: 2
      }
    ]
  };
}

describe("renderMarkdown", () => {
  it("renders required sections and enriched content", () => {
    const c1 = makeCluster("c1", "Topic A");
    const md = renderMarkdown({
      jstDate: "2026-03-02",
      topTopics: [c1],
      rising: [c1],
      deepDiscussion: [c1],
      enriched: [
        {
          clusterId: "c1",
          summaryJa: "要約です",
          tags: ["AI", "Backend"],
          reasonToRead: "読む価値あり"
        }
      ]
    });

    expect(md).toContain("今日のエンジニアトレンド（2026-03-02 JST）");
    expect(md).toContain("🧩 注目トピック Top 7");
    expect(md).toContain("🔥 急上昇 Top 5");
    expect(md).toContain("🧵 議論が深い Top 5");
    expect(md).toContain("要約です");
    expect(md).toContain("#AI #Backend");
    expect(md).toContain("理由: 読む価値あり");
  });

  it("falls back when enrichment is missing", () => {
    const c1 = makeCluster("c1", "Topic A");
    const md = renderMarkdown({
      jstDate: "2026-03-02",
      topTopics: [c1],
      rising: [],
      deepDiscussion: [],
      enriched: []
    });

    expect(md).toContain("要約生成に失敗したため、元リンクを確認してください。");
    expect(md).toContain("増分スコア:+5 / コメント:+3");
  });
});
